src/web/hooks/usePageFilter.js

```diff
@@ -140,7 +144,11 @@ const usePageFilter = (
     rowsPerPage = DEFAULT_ROWS_PER_PAGE;
   }
 
-  if (!returnedFilter.has('rows') && isDefined(rowsPerPage)) {
+  if (
+    !returnedFilter.has('rows') &&
+    isDefined(rowsPerPage) &&
+    rowsPerPage === NO_ROW_LIMIT
```

Comment:
Question: Just for my understanding, what is the reason of adding the check for `rowsPerPage = -1` ?